### PR TITLE
TRUNK-5482 OpenmrsUtil.conceptListHelper(..) to handle duplicates

### PR DESCRIPTION
**Pull request by Am-Coder**
## Description of what I changed

I have added the tests and refractored the OpenmrsUtil.conceptListHelper(..)  method to handle duplicates.

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5482

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java [hunk 1]
```diff
@@ -1003,4 +1006,23 @@ public void applyLogLevels_shouldWarnWhenCalledWithInvalidLevel() {
...
 	}
+	
+	/**
+	 * @see OpenmrsUtil#conceptListHelper(String)
+	 */
+	@Test
+	public void conceptListHelper_shouldNotAllowDuplicateConcepts() {
+		ConceptService conceptService = Context.getConceptService();
+
+		String descriptor = "name:YES | name:NO | name:YES";
```
Comment (sanjulamadurapperuma) [inline api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java:1017]: Could we have this string in the Constants file?

File: api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java [hunk 2]
```diff
@@ -1003,4 +1006,23 @@ public void applyLogLevels_shouldWarnWhenCalledWithInvalidLevel() {
...
+		List<Concept> ret = OpenmrsUtil.conceptListHelper(descriptor);
+		assertEquals(2, ret.size());
+		
+		// both set point to the same concept with id 22		
+		descriptor = "set:30 | set:29";
```
Comment (sanjulamadurapperuma) [inline api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java:1022]: This as well... :)

Comment (Am-Coder) [reply api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java:1022]: This is just a testing string and is specific to this particular test only and not utilized else where in other tests. Will it be fine to declare it as a global constant for all tests of this class? Can there be any reason to declare it as a global String ?

File: api/src/main/java/org/openmrs/util/OpenmrsUtil.java [hunk 3]
```diff
@@ -878,12 +878,17 @@ public static int compareWithNullAsLatest(Date d1, Date d2) {
...
+					conceptSet.addAll(inSet);
 				} else {
-					ret.add(c);
+					conceptSet.add(c);
 				}
 			}
 		}
+		
+		for( Concept c: conceptSet) {
+			ret.add(c);
```
Comment (dilanthas) [inline api/src/main/java/org/openmrs/util/OpenmrsUtil.java:889]: better to use a meaning full name than just 'c'

**Comment by gitcliff** [comment]
@Am-Coder kindly fix the conflicting files,,thanks

**Pull request closed without merging**